### DISCUSSION TITLE
No rendering of decentered surfaces

### DISCUSSION THREAD

**Discussion by Joostth98** [mjhoptics/ray-optics#129] [General]
Dear Micheal Hayford,

When I use decenters in my optical model, the lens surfaces are no longer rendered. The mirrors however are rendered. Is there a reason why this happens? 

Kind regards, 

Joost ter Haar

**Comment by Joostth98** [comment]
Update:

When I 'opm.add_from_file()' the lens is rendered. But when I add a surface with 'sm.add_surface()' it is not rendered.

**Reply by mjhoptics** [reply]
Generally speaking, you can make one or more changes to the model, followed by `opm.update_model()` and that should give the expected result.
If you can attached an example script, that would help me help you further.
Thanks
Mike Hayford
